pkg/kubelet/dockertools/container_gc.go

```diff
@@ -245,35 +281,56 @@ func (cgc *containerGC) GarbageCollect(gcPolicy kubecontainer.ContainerGCPolicy,
 	return nil
 }
 
-func (cgc *containerGC) removeContainer(id string, podNameWithNamespace string, containerName string) {
-	glog.V(4).Infof("Removing container %q name %q", id, containerName)
-	err := cgc.client.RemoveContainer(id, dockertypes.ContainerRemoveOptions{RemoveVolumes: true})
+func (cgc *containerGC) netContainerCleanup(containerInfo containerGCInfo) {
```

Comment:
@dcbw actually, I have a slight concern.

At this point in execution, can we say that the pod that this net container used to belong to is not running? 
 
e.g is this sequence of events possible?
- Pod "nginx-abcdef" is running with infra container X 
- Docker is restarted, infra container goes away.
- Pod "nginx-abcdef" is recreated with a new infra container Y.
- Kubelet (with this PR) tears down infra container X.

I know that Calico has worked around this case for a long time by correlating some information with the namespace/podname (not sure if other plugins do anything similar).  This historically meant that the plugin could tell when a pod has been restarted (and clean up old IP allocations, for example).  However, if the above ordering is correct then the existing plugin will actually delete metadata about a currently running pod :(

It probably would mean changes to the Calico plugin (and any other plugins that might do something similar), but would also render older versions of the plugin incompatible with this change.